src/app/core/services/cluster.service.spec.ts

```diff
@@ -0,0 +1,82 @@
+import {TestBed} from '@angular/core/testing';
+import {MatDialogModule} from '@angular/material/dialog';
+import {AppConfigMockService} from '@test/services/app-config-mock';
+import {AppConfigService} from '@app/config.service';
+import {lastValueFrom} from 'rxjs';
+import {machineDeploymentsFake} from '@test/data/node';
+import {fakeDigitaloceanCluster} from '@test/data/cluster';
+import {fakeProject} from '@test/data/project';
+import {environment} from '@environments/environment';
+import {MatSnackBarModule} from '@angular/material/snack-bar';
+import {RouterTestingModule} from '@angular/router/testing';
+
+describe('ClusterService', () => {
+  let service: ClusterService;
+  let httpController: HttpTestingController;
+
+  const mockMachineDeploymentID = machineDeploymentsFake()[0].id;
+  const mockClusterID = fakeDigitaloceanCluster().id;
+  const mockProjectID = fakeProject().id;
+
+  const newRestRoot = environment.newRestRoot;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [ClusterService, {provide: AppConfigService, useClass: AppConfigMockService}],
+      imports: [HttpClientTestingModule, MatDialogModule, MatSnackBarModule, RouterTestingModule],
+    });
+    httpController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(ClusterService);
+  });
+
+  afterEach(() => {
+    httpController.verify();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should dispatch correct GET request when externalMachineDeploymentNodes() is called', () => {
+    lastValueFrom(service.externalMachineDeploymentNodes(mockProjectID, mockClusterID, mockMachineDeploymentID));
+    const req = httpController.expectOne(
+      `${newRestRoot}/projects/${mockProjectID}/kubernetes/clusters/${mockClusterID}/machinedeployments/${mockMachineDeploymentID}/nodes`
+    );
+    expect(req.request.method).toBe('GET');
+    req.flush({});
+  });
+
+  it('should dispatch correct GET request when externalMachineDeploymentNodesMetrics() is called', () => {
+    lastValueFrom(service.externalMachineDeploymentNodesMetrics(mockProjectID, mockClusterID, mockMachineDeploymentID));
+    const req = httpController.expectOne(
+      `${newRestRoot}/projects/${mockProjectID}/kubernetes/clusters/${mockClusterID}/machinedeployments/${mockMachineDeploymentID}/nodes/metrics`
+    );
+    expect(req.request.method).toBe('GET');
+    req.flush({});
+  });
+
+  it('should dispatch correct GET request when externalMachineDeploymentNodesEvents() is called', () => {
+    lastValueFrom(service.externalMachineDeploymentNodesEvents(mockProjectID, mockClusterID, mockMachineDeploymentID));
+    const req = httpController.expectOne(
+      `${newRestRoot}/projects/${mockProjectID}/kubernetes/clusters/${mockClusterID}/machinedeployments/${mockMachineDeploymentID}/nodes/events`
+    );
+    expect(req.request.method).toBe('GET');
```

Comment:
IMO, we should have shared `enum` for HttpMethods not only it will help to avoid
- spelling mistakes
- It can be used on other test cases too